packages/commercetools/theme/components/Header/HeaderNav.vue

```diff
@@ -0,0 +1,216 @@
+                    </SfLink>
+                  </SfMenuItem>
+                </SfListItem>
+              </SfList>
+            </SfLoader>
+          </SfMegaMenuColumn>
+          <NewCatBanners v-if="!subCategoriesLoading && hasBanners" />
+        </SfMegaMenu>
+      </SfHeaderNavigationItem>
+    </SfHeaderNavigation>
+    <transition name="sf-fade" mode="out-in">
+      <SfMegaMenu
+        v-if="isMobile && isMobileMenuOpen"
+        visible
+        @close="toggleMobileMenu"
+        class="mobile-menu"
+      >
+        <SfMegaMenuColumn
+          v-for="category in categories"
+          :key="category.id"
+          :title="category.name"
+        >
+          <template #title="{ title, changeActive }">
+            <SfMenuItem
+              :label="title"
+              class="sf-mega-menu-column__header"
+              @click="changeActive(title); handleClickCategoryLvl(category.slug, 1)"
+            />
+          </template>
+          <SfLoader :loading="subCategoriesLoading">
+            <div v-if="!subCategoriesLoading">
+              <SfList v-if="hasChildren">
+                <SfListItem
+                  v-for="subCategory in activeCategory[0].children"
+                  :key="subCategory.id"
+                >
+                  <SfMenuItem
+                    :label="subCategory.name"
+                    @click.native="handleClickCategoryLvl(subCategory.slug, 2)"
+                  >
+                    <SfLink>
+                      {{ subCategory.name }}
+                    </SfLink>
+                  </SfMenuItem>
+                </SfListItem>
+              </SfList>
+            </div>
+          </SfLoader>
+          <NewCatBanners v-if="!subCategoriesLoading && hasBanners" />
+        </SfMegaMenuColumn>
+      </SfMegaMenu>
+    </transition>
+  </div>
+</template>
+
+<script>
+import { SfMegaMenu, SfMenuItem, SfList, SfBanner, SfLoader } from '@storefront-ui/vue';
+import { useCategory } from '@vue-storefront/commercetools';
+import { useUiState } from '~/composables';
+import { onSSR } from '@vue-storefront/core';
+import { ref, computed } from '@vue/composition-api';
+import debounce from 'lodash.debounce';
+
+export default {
+  name: 'HeaderNav',
+  components: {
+    SfMegaMenu,
+    SfMenuItem,
+    SfList,
+    SfBanner,
+    SfLoader,
+    NewCatBanners: () => import('./NewCatBanners')
+  },
+  props: {
+    isMobile: {
+      type: Boolean,
+      default: false
+    }
+  },
+  setup (_, { emit, root }) {
+    const { categories, search } = useCategory('menu-categories');
+    const { categories: subCategories, search: subCategoriesSearch, loading: subCategoriesLoading } = useCategory('menu-subCategories');
+    const { toggleMobileMenu, isMobileMenuOpen } = useUiState();
+    const currentCatSlug = ref('');
+    const activeCategory = ref(null);
+    const fetchedSubCategories = ref({});
+    const categoriesWithBanners = ref([
+      { slug: 'new' }
```

Comment:
hardcoded category?

Reply:
Yes, because we don't have any external source of information about the categories in which banners should be visible. Eventually, we can remove banners. WDYT?